### DISCUSSION TITLE
More middleware libraries available on GitHub!

### DISCUSSION THREAD

**Discussion by ALABSTM** [STMicroelectronics/STM32Cube_MCU_Overall_Offer#40] [Announcements]
# More middleware libraries available on GitHub!

**Three** middleware libraries have been recently published on GitHub, **enriching** the already available [list](https://github.com/STMicroelectronics/STM32Cube_MCU_Overall_Offer/blob/master/README.md#stm32cube-middleware-libraries-and-applications) of libraries:
* [stm32-mw-lorawan](https://github.com/STMicroelectronics/stm32-mw-lorawan)
* [stm32-mw-mbedtls](https://github.com/STMicroelectronics/stm32-mw-mbedtls)
* [stm32-mw-mcuboot](https://github.com/STMicroelectronics/stm32-mw-mcuboot)

Still more to come! Enjoy our **STM32Cube** offer and stay tuned!

**Comment by rijesha** [comment]
Are there any instructions on how to start using the stm32-mw-mcuboot library?

It seems like the old implementation is SBSFU Legacy and now ST is moving towards MCUboot. https://wiki.stmicroelectronics.cn/stm32mcu/wiki/Security:SBSFU_by_MCUboot

Can this middleware be directly imported from STM32CubeIDE?

**Reply by ALABSTM** [reply]
Hi @rijesha, 

There are several applications provided within the firmware that show how to use this library. Below a non-exhaustive list:
* [Trusted Firmware for Cortex-M](https://github.com/STMicroelectronics/STM32CubeL5/tree/master/Projects/STM32L562E-DK/Applications/TFM) application provided within the STM32Cube**L5**
* [Secure Boot & Secure Firmware Update](https://github.com/STMicroelectronics/STM32CubeU5/tree/main/Projects/B-U585I-IOT02A/Applications/SBSFU) application provided within the STM32Cube**U5**
* [Root Of Trust](https://github.com/STMicroelectronics/STM32CubeH5/tree/main/Projects/NUCLEO-H503RB/Applications/ROT) application provided within the STM32Cube**H5**

Regarding how to directly import this library from STM32CubeIDE, I suggest that you submit your question to the [ST Community](https://community.st.com/t5/stm32cubeide-mcus/bd-p/stm32-mcu-cubeide-forum). You can also refer to [this](https://community.st.com/t5/stm32-mcus/how-to-add-external-libraries-in-stm32cubeide/ta-p/628619) article.

With regards,
